lib/tbot/config/output.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/lib/tbot/bot"
+	"github.com/gravitational/teleport/lib/tbot/identity"
+)
+
+// Output is an interface that represents configurable Outputs for a bot.
+// These outputs are the core unit of generating artifacts in tbot and are the
+// element users configure to control what is output.
+type Output interface {
```

Comment:
I feel like you may be able to turn this into a struct and then eliminate a lot of the repetition from the various `output_*.go` files. Then have the various `*Output` structs use composition to get the YAML unmarshaling and whatnot and then override anything specific to those particular outputs.